control_unit.py

```diff
@@ -22,22 +22,44 @@ def update_flags(reg_index):
 
 def _BR(instruction):
     """branch"""
-    pass
+    pc_offset = sign_extend((instruction) & 0x1ff, 9)
+    cond_flag = (instruction >> 9) & 0x7
+    if cond_flag & Registers.COND:
```

Comment:
Use `reg_read`, `reg_write` to read/write from/to registers.
This line should be:
`if cond_flag & reg_read(Registers.COND):`